### PR TITLE
Tests for filter.go and comments in types.go

### PR DESCRIPTION
**Pull request by teaguecole**
I accidentally closed my PR... long story, But here it is again with a fix for scotts comment

### CODE REVIEW

File: pkg/apis/servicecatalog/v1beta1/types.go [hunk 1]
```diff
@@ -1314,9 +1316,9 @@ const (
 	// SpecServiceBrokerName is used for ServiceClasses, the parent service broker name.
 	FilterSpecServiceBrokerName = "spec.serviceBrokerName"
 	// SpecClusterServiceClassName is only used for plans, the parent service class name.
-	FilterSpecClusterServiceClassName = "spec.clusterServiceClass.name"
```
Comment (n3wscott) [inline pkg/apis/servicecatalog/v1beta1/types.go]: Please undo this change

File: pkg/apis/servicecatalog/v1beta1/types.go [hunk 2]
```diff
@@ -165,11 +165,13 @@ type CommonServiceBrokerSpec struct {
...
-//
+//   spec.externalName - the value set to [Cluster]ServiceClass.Spec.ExternalName
+//   spec.externalID - the value set to [Cluster]ServiceClass.Spec.ExternalID
 // ServicePlan allowed property names:
-//   name - the value set to [Cluster]ServiceClass.Name
-//   externalName - the value set to [Cluster]ServiceClass.Spec.ExternalName
+//   name - the value set to [Cluster]ServicePlan.Name
+//   spec.externalName - the value set to [Cluster]ServicePlan.Spec.ExternalName
+//   spec.externalID - the value set to [Cluster]ServicePlan.Spec.ExternalID
+//   spec.serviceClassName - the value set to [Cluster]ServicePlan.Spec.ServiceClassRef.Name
```
Comment (n3wscott) [inline pkg/apis/servicecatalog/v1beta1/types.go]: Please change this line to:

```
//   spec.serviceClass.name - the value set to ServicePlan.Spec.ServiceClassRef.Name
//   spec.clusterServiceClass.name - the value set to ClusterServicePlan.Spec.ClusterServiceClassRef.Name
```

File: pkg/apis/servicecatalog/v1beta1/types.go [hunk 3]
```diff
@@ -1314,9 +1316,9 @@ const (
 	// SpecServiceBrokerName is used for ServiceClasses, the parent service broker name.
 	FilterSpecServiceBrokerName = "spec.serviceBrokerName"
 	// SpecClusterServiceClassName is only used for plans, the parent service class name.
-	FilterSpecClusterServiceClassName = "spec.clusterServiceClass.name"
+	FilterSpecClusterServiceClassName = "spec.clusterServiceClassName"
 	// SpecServiceClassName is only used for plans, the parent service class name.
-	FilterSpecServiceClassName = "spec.serviceClass.name"
```
Comment (n3wscott) [inline pkg/apis/servicecatalog/v1beta1/types.go]: And undo this change.

File: pkg/apis/servicecatalog/types.go [hunk 4]
```diff
@@ -144,11 +144,13 @@ type CommonServiceBrokerSpec struct {
...
-//
+//   spec.externalName - the value set to [Cluster]ServiceClass.Spec.ExternalName
+//   spec.externalID - the value set to [Cluster]ServiceClass.Spec.ExternalID
 // ServicePlan allowed property names:
-//   name - the value set to [Cluster]ServiceClass.Name
-//   externalName - the value set to [Cluster]ServiceClass.Spec.ExternalName
+//   name - the value set to [Cluster]ServicePlan.Name
+//   spec.externalName - the value set to [Cluster]ServicePlan.Spec.ExternalName
+//   spec.externalID - the value set to [Cluster]ServicePlan.Spec.ExternalID
+//   spec.serviceClassName - the value set to [Cluster]ServicePlan.Spec.ServiceClassRef.Name
```
Comment (n3wscott) [inline pkg/apis/servicecatalog/types.go]: Same comments apply to here from the other types file

**Comment by teaguecole** [comment]
@carolynvs put on those kicking boots please, this build needs a kick. (Relates to #2036)

**Comment by carolynvs** [comment]
Kicked!
